katz/src/main/kotlin/katz/instances/NonEmptyListMonad.kt

```diff
@@ -11,18 +11,19 @@ interface NonEmptyListMonad : Monad<NonEmptyList.F> {
             fa.ev().flatMap { f(it).ev() }
 
     @Suppress("UNCHECKED_CAST")
-    private tailrec fun <A, B> go(buf: ArrayList<B>, f: (A) -> HK<NonEmptyList.F, Either<A, B>>, v: NonEmptyList<Either<A, B>>): Unit =
-            when (v.head) {
-                is Either.Right<*> -> {
-                    buf += v.head.b as B
-                    val x = NonEmptyList.fromList(v.tail)
-                    when (x) {
-                        is Option.Some<NonEmptyList<Either<A, B>>> -> go(buf, f, x.value)
-                        is Option.None -> Unit
-                    }
+    private tailrec fun <A, B> go(buf: ArrayList<B>, f: (A) -> HK<NonEmptyList.F, Either<A, B>>, v: NonEmptyList<Either<A, B>>) {
+        when (v.head) {
+            is Either.Right<*> -> {
+                buf += v.head.b as B
+                val x = NonEmptyList.fromList(v.tail)
+                when (x) {
+                    is Option.Some<NonEmptyList<Either<A, B>>> -> go(buf, f, x.value)
+                    is Option.None -> Unit
                 }
-                is Either.Left<*> -> go(buf, f, NonEmptyList.fromListUnsafe(f(v.head.a as A).ev().all + v.tail))
             }
+            is Either.Left<*> -> go(buf, f, NonEmptyList.fromListUnsafe(f(v.head.a as A).ev().all + v.tail))
+        }
+    }
```

Comment:
No. Removed the need to return something on the body. I followed Fernando's advice to avoid compilation errors.